application/src/main/java/com/musicforall/services/playlist/PlaylistServiceImpl.java

```diff
@@ -66,14 +67,24 @@ public void delete(Integer playlistId) {
     }
 
     @Override
+    public void addTrack(Integer playlistId, Integer trackId) {
+
+        final Playlist playlist = dao.get(Playlist.class, playlistId);
+        final Track track = dao.get(Track.class, trackId);
+        playlist.getTracks().add(track);
+    }
+
+    @Override
     public void addTracks(Integer playlistId, Set<Track> tracks) {
         final Playlist playlist = dao.get(Playlist.class, playlistId);
         playlist.addTracks(tracks);
         save(playlist);
-
     }
 
-    public void setDao(Dao dao) {
-        this.dao = dao;
+    @Override
+    public void removeTrack(Integer playlistId, Integer trackId) {
+
+        final Playlist playlist = dao.get(Playlist.class, playlistId);
+        playlist.getTracks().removeIf(track -> Objects.equals(track.getId(), trackId));
     }
```

Comment:
Perhaps you quoted wrong lines of code, because there is no track saving happens in this method. Could you elaborate on this, please?